PacificPower/readPP.js

```diff
@@ -70,7 +70,98 @@ let continueVarMonthly = 0;
 let page = "";
 let pp_meter_id = "";
 
+let pp_meters_exclusion_list = null;
+let pp_meters_exclude = [];
+let pp_meters_include = [];
+let pp_meters_exclude_not_found = [];
+
+function compareMeterAgainstExclusionList(PPTable) {
+  const meter = pp_meters_exclusion_list.find(
+    (meter) => meter.pp_meter_id === PPTable.pp_meter_id,
+  );
+
+  if (!meter) {
+    console.log(
+      `Meter ${PPTable.pp_meter_id} is not in the exclusion list: NEW METER`,
+    );
+    pp_meters_exclude_not_found.push(PPTable.pp_meter_id);
+    PPArray.push(PPTable);
+    return;
+  }
+
+  switch (meter.status) {
+    case "exclude":
+      console.log(`Meter ${PPTable.pp_meter_id} is excluded from db`);
+      pp_meters_exclude.push(PPTable.pp_meter_id);
+      break;
+    case "include":
+      console.log(`Meter ${PPTable.pp_meter_id} is included in db`);
+      pp_meters_include.push(PPTable.pp_meter_id);
+      PPArray.push(PPTable);
+      break;
+    case "new":
+      console.log(
+        `Meter ${PPTable.pp_meter_id} status needs updating, include in db for now.`,
+      );
+      pp_meters_include.push(PPTable.pp_meter_id);
+      PPArray.push(PPTable);
+      break;
+    default:
+      console.log(`Meter ${PPTable.pp_meter_id} unrecognized status`);
+  }
+}
+
+async function addNewMetersToDatabase() {
+  for (let i = 0; i < pp_meters_exclude_not_found.length; i++) {
+    await axios({
+      method: "post",
+      url: `${process.env.DASHBOARD_API}/ppupload`,
+      data: {
+        id: pp_meters_exclude_not_found[i],
+        pwd: process.env.API_PWD,
+      },
+    })
+      .then((res) => {
+        console.log(
+          `RESPONSE: ${res.status}, TEXT: ${res.statusText}, DATA: ${res.data}`,
+        );
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+}
+
 (async () => {
+  // fetch the meter exclusion list
+  pp_meters_exclusion_list = await axios({
+    method: "get",
+    url: `${process.env.DASHBOARD_API}/ppexclude`,
+  })
+    .then((res) => {
+      if (res.status < 200 || res.status >= 300) {
+        throw new Error("Failed to fetch PP Meter Exclusion List");
+      }
+
+      console.log(
+        `RESPONSE: ${res.status}, TEXT: ${
+          res.statusText
+        }, DATA: ${JSON.stringify(res.data)}`,
+      );
+      return res.data;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+
+  if (pp_meters_exclusion_list) {
+    console.log("PP Meter Exclusion List: ", pp_meters_exclusion_list);
```

Comment:
Let's change this to something like:

```js
    for (let i = 0; i < pp_meters_exclusion_list.length; i++) {
      console.log(pp_meters_exclusion_list[i]);
    }
 ```
 
 Since JS arrays / lists get cut off in log statements once they reach a certain size (will get `...4 more items` etc.)

Reply:
This was for debugging that I forgot to remove, I've changed it to only log how many meters are in the list now to indicate that it was successfully retrieved.

Reply:
LGTM